wordpress-mobile/WordPress-iOS
# Remove UISplitViewController from the new Me screen

#### The new Me screen does not use `UISplitViewController` anymore.
- Default modal presentation style for iPhone
- Form sheet presentation style for iPad
- is the app is visualized in a portion of a split screen on iPad with compact horizontal size class, the modal presentation will be the same as an iPhone
- State restoration reflects what was in place with the previous Me tab screen (only at the top level). If we feel that we need to restore other levels, we can handle it in a separate PR.
- The `Done` button is visible at the root level and the first sub level of navigation.
- Took the occasion to reorganize files in Classes->ViewRelated->Me since I was working on those files
##### Default
![iPhoneLandscape](https://user-images.githubusercontent.com/34376330/75264861-a8bee000-57b5-11ea-9b9e-4395506b9c6f.gif)
 ##### Form Sheet
![iPadSplitScreen](https://user-images.githubusercontent.com/34376330/75264878-af4d5780-57b5-11ea-9d63-72994699b91c.gif)


Ref #13319

To test:
- Go to `My Site`
- tap the `Me` button

on iPhone with horizontal regular size class (e.g. 11 Pro Max)
- check that the view in portrait and landscape do not have split view
- check that the done button is visible and actionable at the root and first sub level

on iPad
- check that the presentation style is form sheet, and the done button is visible and actionable at root and first sub level

State restoration - on any device
- Tap the `Me` button to open the `Me` screen
- Go to the device or simulator home screen
- Stop the app from Xcode
- Relaunch from Xcode and verify that the `Me` screen shows up



PR submission checklist:
